documentation/general/dotnet-run-file-multi-file.md

```diff
@@ -0,0 +1,136 @@
+# Multi-file support in `dotnet run app.cs`
+
+## Scenarios
+
+Some multi-file scenarios to consider:
+
+### Direcotry-based apps
+
+- Directory of files that are all part of the same app
+- Only a single entry-point file
+- Might be using differnt SDKs, e.g. `Microsoft.NET.Sdk` vs. `Microsoft.NET.Sdk.Web`, that have different default items
+
+Example console app:
+
+```text
+- myapp/
+  - myapp.cs
+  - helpers.cs
+```
+
+Example Web API app:
+
+```text
+- webapi/
+  - webapi.cs
+  - helpers.cs
+  - customerapi.cs
+  - productapi.cs
+  - appsettings.json
+```
+
+Example Razor app:
+
+```text
+- razorapp/
+  - razorapp.cs
+  - helpers.cs
+  - Pages/
+    - _ViewImports.razor
+    - Index.razor
+    - App.razor
+    - Layout.razor
+  - appsettings.json
+```
+
+Example Aspire app host app:
+
+```text
+- apphost/
+  - apphost.cs
+  - helpers.cs
+  - appsettings.json
+```
+
+### Directories with multiple separate entry-point files
+
+- Each entry-point file is a separate app
+- Some entry-point files require types from some non-entry-point files
```

Comment:
`csi.exe` from msbuild distribution uses `.csx` file extension for executable C# scripts. Maybe this file extension could be used to distinguish easily entry-point and non-entry-point files?